src/Workspaces/Core/Portable/Formatting/AbstractFormattingService.cs

```diff
@@ -14,7 +15,10 @@ namespace Microsoft.CodeAnalysis.Formatting
     /// </summary>
     internal abstract class AbstractFormattingService : IFormattingService
     {
-        public Task<Document> FormatAsync(Document document, IEnumerable<TextSpan>? spans, SyntaxFormattingOptions options, CancellationToken cancellationToken)
-            => Formatter.FormatAsync(document, spans, options, rules: null, cancellationToken);
+        public Task<Document> FormatAsync(Document document, IEnumerable<TextSpan>? spans, LineFormattingOptions lineFormattingOptions, SyntaxFormattingOptions? syntaxFormattingOptions, CancellationToken cancellationToken)
+        {
+            Contract.ThrowIfNull(syntaxFormattingOptions);
```

Comment:
Can you put `[NotNull]` on the parameter? Or does it not work/make the compiler complain?

Reply:
NotNull does not have an effect on the method body. It "Specifies that an input argument was not null when the call returns."